cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/BoardsConfigService.kt

```diff
@@ -30,11 +36,16 @@ class BoardsConfigService(
         contentRepository.save(widgetId, content)
     }
 
+    fun setDeletionHelper(helper: BoardDeletionHelper) {
+        boardDeletionHelper = helper
+    }
+
     private fun getWidgetById(boardsConfig: JsonObject) =
             boardsConfig.getJsonObject("widgets")
                     ?.getJsonObject("widgetsById") ?: JsonObject()
 
     private fun clearContent(widgetId: String, widget: JsonObject) {
+        contentRepository.delete(widgetId)
```

Comment:
Added it here instead of into `WidgetRuntimeService` (which handles widget deletions), mostly since we already have here a method that is intended for clearing content and we have access to `contentRepository`.

Reply:
In my opinion this is a bad place for this method - this makes the method do 2 unrelated things:

- removes content node before save - we don't want a content to be saved with config - it is saved somewhere else
- and now deletes widget - i think this might be a regression here

Reply:
Fixed. Added different solution that manages all widgets in whole config, deleting no longer used and emitting purge events for these as well.